src/rand.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Authors (alpha order): Adrian Catangiu @acatangiu, Alexandra Lordache @aghecenco, Alexandru Agache @alexandruag, Diana Popa @dianpopa, Matt Forrester @forbesmyester, Radu Weiss @raduweiss
+
+extern crate libc;
+use std::ffi::OsString;
+use std::str;
+
+pub fn timestamp_cycles() -> u64 {
+    #[cfg(target_arch = "x86_64")]
+    // Safe because there's nothing that can go wrong with this call.
+    unsafe {
+        std::arch::x86_64::_rdtsc() as u64
+    }
+    #[cfg(not(target_arch = "x86_64"))]
+    {
+        let mut ts = libc::timespec {
+            tv_sec: 0,
+            tv_nsec: 0,
+        };
+
+        unsafe {
+            libc::clock_gettime(libc::CLOCK_MONOTONIC, &mut ts);
+        }
+        (ts.tv_sec as u64) * 1000000000 + (ts.tv_nsec as u64)
```

Comment:
Can you add a define here instead of 1000000000? Also, clippy will complain if you use large numbers without formatting. It should probably be changed to `1_000_000_000`.

Reply:
Yep